gaphor/SysML/diagramlabel.py

```diff
@@ -0,0 +1,27 @@
+from gaphor.diagram.diagramlabel import diagram_label
+from gaphor.SysML.sysml import SysMLDiagram, Block, ConstraintBlock, Requirement
+from gaphor.UML.uml import Activity, Package, Interaction, Profile, StateMachine
+
+
+@diagram_label.register(SysMLDiagram)
+def sysml_diagram_label(diagram):
+    def element_type(el):
+        if isinstance(el, Activity):
+            return "activity"
+        if isinstance(el, ConstraintBlock):
+            return "constraint block"
+        if isinstance(el, Block):
+            return "block"
+        if isinstance(el, Interaction):
+            return "interaction"
+        if isinstance(el, Profile):
+            return "profile"
+        if isinstance(el, Package):
+            return "package"
+        if isinstance(el, Requirement):
+            return "requirement"
+        if isinstance(el, StateMachine):
+            return "state machine"
+        return ""
+
+    return f"[{element_type(diagram.element)}] {diagram.element.name} [{diagram.name}]"
```

Comment:
I ran into one issue: when a diagram does not have a parent (`diagram.element`), this raises an error.

```suggestion
    if diagram.element:
        return f"[{element_type(diagram.element)}] {diagram.element.name} [{diagram.name}]"
    else:
        return diagram.name
```

Reply:
Good catch! According to the SysML spec, diagrams have to have a parent. At the moment, we do not (yet?) have these constraints, so we can create diagrams wherever we like (technically we have few constraints on this, but still, very loose). Your proposed default makes sense.